akd/configs/data_search_config.py

```diff
@@ -0,0 +1,266 @@
+"""
+Centralized configuration for data search components.
+"""
+
+import os
+from enum import Enum
+from typing import Optional
+
+from pydantic import BaseModel, Field, HttpUrl
+
+
+class Environment(str, Enum):
+    """Supported environments."""
+
+    DEVELOPMENT = "development"
+    STAGING = "staging"
+    PRODUCTION = "production"
+
+
+class WebSocketConfig(BaseModel):
+    """WebSocket connection configuration."""
+
+    host: str = Field(default="localhost", description="WebSocket host")
+    port: int = Field(default=8003, description="WebSocket port")
+    max_reconnect_attempts: int = Field(
+        default=5,
+        description="Maximum reconnection attempts",
+    )
+    base_reconnect_delay: int = Field(
+        default=1000,
+        description="Base reconnection delay in milliseconds",
+    )
+    connection_timeout: float = Field(
+        default=10.0,
+        description="Connection timeout in seconds",
+    )
+
+    @property
+    def url(self) -> str:
+        """Get the WebSocket URL."""
+        return f"ws://{self.host}:{self.port}"
+
+    @property
+    def http_base_url(self) -> str:
+        """Get the HTTP base URL."""
+        return f"http://{self.host}:{self.port}"
+
+
+class MCPConfig(BaseModel):
+    """MCP server configuration."""
+
+    endpoint: HttpUrl = Field(
+        default="http://localhost:8080/mcp/cmr/mcp/",
+        description="MCP server endpoint URL",
+    )
+    timeout_seconds: float = Field(
+        default=30.0,
+        description="Request timeout in seconds",
+    )
+    max_retries: int = Field(default=3, description="Maximum number of retry attempts")
+    retry_delay: float = Field(
+        default=1.0,
+        description="Base delay between retries in seconds",
+    )
+
+
+class LLMConfig(BaseModel):
+    """LLM service configuration."""
+
+    api_key: Optional[str] = Field(default=None, description="OpenAI API key")
+    model_name: str = Field(default="gpt-4", description="Model name to use")
+    temperature: float = Field(default=0.1, description="Model temperature")
+    max_retries: int = Field(
+        default=3,
+        description="Maximum retry attempts for LLM calls",
+    )
+    base_delay: float = Field(
+        default=1.0,
+        description="Base delay for exponential backoff",
+    )
```

Comment:
There's arelady `akd.agents._base.BaseAgentConfig`. Reuse that or inherit from that.